src/commands/scaling/addScaleRule/queue/QueueLengthStep.ts

```diff
@@ -0,0 +1,35 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import { localize } from '../../../../utils/localize';
+import { IAddScaleRuleWizardContext } from '../IAddScaleRuleWizardContext';
+
+export class QueueLengthStep extends AzureWizardPromptStep<IAddScaleRuleWizardContext> {
+    public async prompt(context: IAddScaleRuleWizardContext): Promise<void> {
+        context.queueLength = Number((await context.ui.showInputBox({
+            prompt: localize('queueLengthPrompt', 'Enter a queue length.'),
+            validateInput: async (value: string | undefined): Promise<string | undefined> => await this.validateInput(value)
+        })).trim());
+    }
+
+    public shouldPrompt(context: IAddScaleRuleWizardContext): boolean {
+        return context.queueLength === undefined;
+    }
+
+    private async validateInput(length: string | undefined): Promise<string | undefined> {
+        length = length ? length.trim() : '';
+
+        const thirtyTwoBitMaxSafeInteger = 2147483647;
+        if (!/^[1-9]+[0-9]*$/.test(length)) {
```

Comment:
Could probably make this regex a constant. 

Or if `HttpConcurrentRequestStep` actually follows `thirtyTwoBitMaxSafeInteger` as well, maybe you could make a base step that has this  implementation of `validateInput`